File: app/src/main/java/com/github/gmetal/presentation/model/TvShowModel.java
package com.github.gmetal.presentation.model;

import org.parceler.Parcel;

import java.util.ArrayList;
import java.util.List;

@Parcel
public class TvShowModel extends MediaInfoModel {

    protected List<TvShowSeasonModel> seasons;

    public TvShowModel() {

        super();
        seasons = new ArrayList<>();
    }

    @Override
    public String getMediaId() {
        return getId();
    }

    @Override
    public String getImage() {
        return imageURL;
    }

    @Override
    public String getTitle() {
        return name;
    }

    public List<TvShowSeasonModel> getSeasons() {
        return seasons;
    }

    public void setSeasons(final List<TvShowSeasonModel> seasons) {

        this.seasons = seasons;
    }
}

File: app/src/main/java/com/github/gmetal/presentation/model/TvShowModel.kt
package com.github.gmetal.presentation.model

import android.os.Parcelable
import kotlinx.android.parcel.Parcelize

@Parcelize
class TvShowModel(
    override var overview: String = "",
    override var voteCount: Int = 0,
    override var voteAverage: Float = 0.0f,
    override var id: String = "",
    override var name: String = "",
    override var imageURL: String = "",
    override var isLoadingObject: Boolean = false,
    var seasons: List<TvShowSeasonModel> = emptyList()
) : MediaInfoModel, Parcelable {

    override val mediaId: String
        get() = id

    override val image: String
        get() = imageURL

    override val title: String
        get() = name

    init {
        seasons = ArrayList()
    }
}
